Error providing `aws.allowedAccountIds` in `sst.config.ts`

In particular setting `providers.aws.allowedAccountIds` in `Config.app` of `sst.config.ts` which has typescript type `$util.Input<$util.Input<string>[]>` which is basically `string[]`  gives 

```
panic: json: cannot unmarshal array into Go struct field App.providers of type string

goroutine 1 [running]:
main.main()
    /home/runner/work/ion/ion/cmd/sst/main.go:659 +0xa88

```

Looks like it's to do with the  struct for app providers only allowing string values in the following file. https://github.com/sst/ion/blob/955765ad6e77b58e1e20d44c72e91e028975d051/pkg/project/project.go#L19

--- a/pkg/project/project.go
+++ b/pkg/project/project.go
@@ -13,10 +13,10 @@ import (
 )
 
 type App struct {
-	Name          string                       `json:"name"`
-	Stage         string                       `json:"stage"`
-	RemovalPolicy string                       `json:"removalPolicy"`
-	Providers     map[string]map[string]string `json:"providers"`
+	Name          string                            `json:"name"`
+	Stage         string                            `json:"stage"`
+	RemovalPolicy string                            `json:"removalPolicy"`
+	Providers     map[string]map[string]interface{} `json:"providers"`
 }
 
 type Project struct {
@@ -140,7 +140,7 @@ console.log("~j" + JSON.stringify(mod.app({
 		proj.app.Stage = input.Stage
 
 		if proj.app.Providers == nil {
-			proj.app.Providers = map[string]map[string]string{}
+			proj.app.Providers = map[string]map[string]interface{}{}
 		}
 
 		if proj.app.Name == "" {
@@ -161,7 +161,7 @@ console.log("~j" + JSON.stringify(mod.app({
 
 func (proj *Project) LoadProviders() error {
 	if _, ok := proj.app.Providers["aws"]; !ok {
-		proj.app.Providers["aws"] = map[string]string{}
+		proj.app.Providers["aws"] = map[string]interface{}{}
 	}
 
 	proj.Providers = map[string]provider.Provider{}

--- a/pkg/project/provider/aws.go
+++ b/pkg/project/provider/aws.go
@@ -24,7 +24,7 @@ import (
 )
 
 type AwsProvider struct {
-	args        map[string]string
+	args        map[string]interface{}
 	config      aws.Config
 	bootstrap   *awsBootstrapData
 	credentials sync.Once
@@ -146,7 +146,7 @@ func (a *AwsProvider) Cancel(app string, stage string) error {
 
 const BOOTSTRAP_VERSION = 1
 
-func (a *AwsProvider) Init(app string, stage string, args map[string]string) (err error) {
+func (a *AwsProvider) Init(app string, stage string, args map[string]interface{}) (err error) {
 	a.args = args
 
 	cfg, err := a.resolveConfig()
@@ -304,11 +304,11 @@ func (a *AwsProvider) resolveConfig() (aws.Config, error) {
 	cfg, err := config.LoadDefaultConfig(
 		ctx,
 		func(lo *config.LoadOptions) error {
-			if a.args["profile"] != "" {
-				lo.SharedConfigProfile = a.args["profile"]
+			if profile, ok := a.args["profile"].(string); ok && profile != "" {
+				lo.SharedConfigProfile = profile
 			}
-			if a.args["region"] != "" {
-				lo.Region = a.args["region"]
+			if region, ok := a.args["region"].(string); ok && region != "" {
+				lo.Region = region
 				lo.DefaultRegion = "us-east-1"
 			}
 			return nil

--- a/pkg/project/provider/cloudflare.go
+++ b/pkg/project/provider/cloudflare.go
@@ -9,7 +9,7 @@ import (
 type CloudflareProvider struct {
 }
 
-func (c *CloudflareProvider) Init(app string, stage string, args map[string]string) error {
+func (c *CloudflareProvider) Init(app string, stage string, args map[string]interface{}) error {
 	return nil
 }
 

--- a/pkg/project/provider/provider.go
+++ b/pkg/project/provider/provider.go
@@ -44,7 +44,7 @@ func (dt *DevTransport) Publish(input interface{}) error {
 }
 
 type Provider interface {
-	Init(app, stage string, provider map[string]string) error
+	Init(app, stage string, provider map[string]interface{}) error
 }
 
 type DevSession interface {

--- a/pkg/project/stack.go
+++ b/pkg/project/stack.go
@@ -233,7 +233,14 @@ func (s *stack) Run(ctx context.Context, input *StackInput) error {
 			if provider == "cloudflare" && key == "accountId" {
 				continue
 			}
-			config[fmt.Sprintf("%v:%v", provider, key)] = auto.ConfigValue{Value: value}
+			switch v := value.(type) {
+			case string:
+				config[fmt.Sprintf("%v:%v", provider, key)] = auto.ConfigValue{Value: v}
+			case []string:
+				for i, val := range v {
+					config[fmt.Sprintf("%v:%v[%d]", provider, key, i)] = auto.ConfigValue{Value: val}
+				}
+			}
 		}
 	}
 	err = stack.SetAllConfig(ctx, config)
@@ -469,7 +476,14 @@ func (s *stack) Import(ctx context.Context, input *ImportOptions) error {
 			if provider == "cloudflare" && key == "accountId" {
 				continue
 			}
-			config[fmt.Sprintf("%v:%v", provider, key)] = auto.ConfigValue{Value: value}
+			switch v := value.(type) {
+			case string:
+				config[fmt.Sprintf("%v:%v", provider, key)] = auto.ConfigValue{Value: v}
+			case []string:
+				for i, val := range v {
+					config[fmt.Sprintf("%v:%v[%d]", provider, key, i)] = auto.ConfigValue{Value: val}
+				}
+			}
 		}
 	}
 	err = stack.SetAllConfig(ctx, config)

